nevergrad/functions/images/imagelosses.py

```diff
@@ -1,35 +1,102 @@
+import cv2
+import imquality.brisque as brisque
+import lpips
 import os
+import torch
 import typing as tp
 import numpy as np
 from nevergrad.functions.base import UnsupportedExperiment
+from nevergrad.common.decorators import Registry
+
+
+registry: Registry[tp.Any] = Registry()
 
 
 class ImageLoss:
     def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         pass
 
     def __call__(self, img: np.ndarray) -> float:
-        raise NotImplementedError
+        raise NotImplementedError(f"__call__ undefined in class {type(self)}")
 
 
-class SumAbsoluteDifferences(ImageLoss):
+class ImageLossWithReference(ImageLoss):
     def __init__(self, reference: np.ndarray) -> None:
         if reference is None:
             raise ValueError("A reference is required")
         self.reference = reference
         super().__init__(reference)
+        assert len(self.reference.shape) == 3, self.reference.shape
         self.domain_shape = self.reference.shape
 
+
+@registry.register
+class SumAbsoluteDifferences(ImageLossWithReference):
+    def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
+        super().__init__(reference)
+
     def __call__(self, x: np.ndarray) -> float:
         assert x.shape == self.domain_shape, f"Shape = {x.shape} vs {self.domain_shape}"
         value = float(np.sum(np.fabs(x - self.reference)))
         return value
 
 
+@registry.register
+class LpipsAlex(ImageLossWithReference):
+    def __init__(self, reference: np.ndarray) -> None:
+        super().__init__(reference)
+        self.loss_fn = lpips.LPIPS(net="alex")
+
+    def __call__(self, img: np.ndarray) -> float:
+        assert img.shape[2] == 3
+        assert len(img.shape) == 3
+        img0 = np.expand_dims(img, 0)
+        assert img0.shape[3] == 3
+        assert img0.shape[0] == 1
+        assert len(img0.shape) == 4
+        img1 = np.expand_dims(self.reference, 0)
+        img0 = torch.clamp(torch.Tensor(img0).permute(0, 3, 1, 2) / 255.0, 0, 1) * 2.0 - 1.0
+        img1 = torch.clamp(torch.Tensor(img1).permute(0, 3, 1, 2) / 255.0, 0, 1) * 2.0 - 1.0
+        return self.loss_fn(img0, img1)
+
+
+@registry.register
+class LpipsVgg(LpipsAlex):
+    def __init__(self, reference: np.ndarray) -> None:
+        super().__init__(reference)
+        self.loss_fn = lpips.LPIPS(net="vgg")
+
```

Comment:
I think it would make more sense to have a class Lpips and two classes inheriting from it for Alexnet and VGG
```suggestion
class Lpips(ImageLossWithReference):
    def __init__(self, reference: np.ndarray, net: str) -> None:
        super().__init__(reference)
        self.loss_fn = lpips.LPIPS(net=net)

    def __call__(self, img: np.ndarray) -> float:
        assert img.shape[2] == 3
        assert len(img.shape) == 3
        img0 = np.expand_dims(img, 0)
        assert img0.shape[3] == 3
        assert img0.shape[0] == 1
        assert len(img0.shape) == 4
        img1 = np.expand_dims(self.reference, 0)
        img0 = torch.clamp(torch.Tensor(img0).permute(0, 3, 1, 2) / 255.0, 0, 1) * 2.0 - 1.0
        img1 = torch.clamp(torch.Tensor(img1).permute(0, 3, 1, 2) / 255.0, 0, 1) * 2.0 - 1.0
        return self.loss_fn(img0, img1)


@registry.register
class LpipsVgg(Lpips):
    def __init__(self, reference: np.ndarray) -> None:
        super().__init__(reference, net="vgg")


@registry.register
class LpipsAlex(Lpips):
    def __init__(self, reference: np.ndarray) -> None:
        super().__init__(reference, net="alex")
```

Reply:
Right: done.